lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -695,4 +781,42 @@ private Function<Integer, Object> getConverter(PrimitiveType primitiveType)
             super(columnPath, Integer.class);
         }
     }
+
+    private sealed interface ValuesList
+            permits StatisticsRangeList, DiscreteValuesList
```

Comment:
`sealed` and `permits` are cool, but pretty redundant, since all these classes are private to this file anyway